apps/api/src/app/events/usecases/send-message/send-message-email.usecase.ts

```diff
@@ -283,6 +293,18 @@ export class SendMessageEmail extends SendMessageBase {
     await this.sendErrors(email, integration, message, command, notification);
   }
 
+  private async getReplyTo({ environmentId, transactionId }: { environmentId: string; transactionId: string }) {
+    const environment = await this.environmentRepository.findOne({ _id: environmentId });
+
+    let replyCallbackRoute = process.env.REPLY_CALLBACK_ROUTE;
+
+    if (environment.dns?.mxRecordConfigured) {
+      replyCallbackRoute = environment.dns?.domain;
+    }
+
+    return `parse+${transactionId}@${replyCallbackRoute}`;
```

Comment:
Can we add the environment id here as well to make sure this is uniquely identifiable?